src/averageAges.js

```diff
@@ -1,5 +1,42 @@
 'use strict';
 
+const getByKeyValue = (key, value) => people => {
+  const peopleCopy = [ ...people ];
+
+  return peopleCopy.filter(person => person[key] === value);
+};
+
+const getMen = getByKeyValue('sex', 'm');
+const getWomen = getByKeyValue('sex', 'f');
+
+const getValuesArr = (key) => people => {
+  const resultArr = [];
+
+  people.filter(person => person[key] !== null
+    ? resultArr.push(person[key]) : null);
+
+  return resultArr;
+};
+
+const getMotherNames = getValuesArr('mother');
+
+const getMothers = (people, group = people) => {
+  const women = getWomen(people);
+  const motherNames = getMotherNames(group);
```

Comment:
You pass `group` into `getMotherNames`. But `getMotherNames` does not take parameters :)

Reply:
![image](https://user-images.githubusercontent.com/120763734/221141378-ff7ed70d-3124-4e97-a3d7-18861626f785.png)
please be more attentive too

I have callback constructor `getValuesArr = (key) =>` that returns function `people => {}`, wose accept some array (called people) and reterns array of people

`getMotherNames(group)` is example of function, that was created by fuction constructor
and designed to accept an array of objects of people and return an array of string-kind names of mother

so consider written above and
thank you for your comments